tests/unit/core/utils/test_compatibility.js

```diff
@@ -935,4 +939,138 @@ describe(__filename, () => {
       expect(isQuantumCompatible({ addon })).toEqual(true);
     });
   });
+
+  describe('correctedLocationForPlatform', () => {
+    const _correctedLocationForPlatform = ({
+      clientApp,
+      location = createFakeLocation(),
+      userAgentInfo,
+    }) => {
+      return correctedLocationForPlatform({
+        clientApp,
+        location,
+        userAgentInfo,
+      });
+    };
+
+    it('returns a link with `CLIENT_APP_ANDROID` replaced with `CLIENT_APP_FIREFOX` when on Firefox desktop', () => {
+      const pathname = `/en-US/${CLIENT_APP_ANDROID}/addon/slug/`;
+      const search = '?src=featured';
+
+      expect(
+        _correctedLocationForPlatform({
+          clientApp: CLIENT_APP_ANDROID,
+          location: createFakeLocation({ pathname, search }),
+          userAgentInfo: {
```

Comment:
I started with tests like that, but then I thought it might be odd to just test one particular user agent when there are many for which the condition should be true, so I figured just specifying the data we cared about made more sense. But I am fine to switch back to using user agent strings, so I'll make that change.